gateway-ha/src/main/java/io/trino/gateway/ha/resource/TrinoResource.java

```diff
@@ -125,7 +125,7 @@ public Response readAllSelectors(@QueryParam("useSchema")
 
   @GET
   @Path("/selector/read/{resourceGroupId}")
-  public Response readSelector(@QueryParam("resourceGroupId") String resourceGroupIdStr,
+  public Response readSelector(@PathParam("resourceGroupId") String resourceGroupIdStr,
                                @QueryParam("useSchema") String useSchema) {
     if (Strings.isNullOrEmpty(resourceGroupIdStr)) { // if query not specified, return all
```

Comment:
Trivial. How about cleaning comments or rewrite it so that it means resource group string instread of query? Same applies for line 64 96.

Reply:
Actually I think we don't have to handle the `if (Strings.isNullOrEmpty(resourceGroupIdStr))` case here (so the comment is not needed either). `{resourceGroupId}` will have to match the `[^/]+` regex:

>  If regex is not supplied then a default value of [^/]+ which terminates at a path segment boundary, is used.

https://jakarta.ee/specifications/restful-ws/3.0/apidocs/jakarta/ws/rs/path

`/globalproperty/delete/{name}` below doesn't have that condition, and `POST /globalproperty/delete` and `POST /globalproperty/delete/` both return a normal 404.

Reply:
Perfect.